### PR TITLE
Add drift detection between certificates and secrets for certain apps

### PR DESCRIPTION
**Pull request by njuettner**
Towards: https://github.com/giantswarm/giantswarm/issues/29024

As mentioned in the issue, this introduces a new alert on missing
Certificate CR's.

For now I only added `hubble-ui*` and `aws-admission-controller` as a
test.

Here's an example where we would get paged, `anteater` misses the certificate for `aws-admission-controller`, secret is present but `cert-manager` wouldn't renew it:

image

### Checklist

- [ ] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

**Comment by Gacko** [comment]
So this checks `tls.crt`s in secrets and looks for their corresponding certificate CR just by name, right? What about certificates which set a different secret name?

**Comment by njuettner** [comment]
@Gacko correct, unfortunately it makes it a bit fragile. But almost every app which I checked does have the same naming besides apps which do create certificates different like Kyverno but that was never an issue.

I thought about extending the cert-exporter to add some information like the app name but there simply no information on the resource itself where we could get it.

**Comment by Gacko** [comment]
Normally and if a secret is managed by cert-manager, it sets some annotations. So you can tell if a secret is managed by a certificate by its annotations, IIRC.

**Comment by njuettner** [comment]
@Gacko good input, I created a [PR](https://github.com/giantswarm/cert-exporter/pull/331) for `cert-exporter` to add an additional field for `cert_exporter_secret_...` adding the annotation there which allows me to change the alert slighty.

**Comment by njuettner** [comment]
PR is merged for `cert-exporter` I slightly changed the PromQL so it doesn't rely on the same name for secret and certificate.

image

**Review by Gacko — APPROVED** [review]
My PromQL skills aren't that good, but I guess it's fine, especially because you implemented and tested the suggestions. 😬

File: helm/prometheus-rules/templates/alerting-rules/certificate.management-cluster.rules.yml [hunk 1]
```diff
@@ -23,6 +23,18 @@ spec:
         severity: page
         team: rocket
         topic: security
+    - alert: ManagementClusterCertificateIsMissing
+      annotations:
+        description: '{{`Cannot renew Certificate for Secret {{ $labels.namespace }}/{{ $labels.name }} on {{ $labels.cluster_id }} because it is missing.`}}'
+        opsrecipe: managed-app-cert-manager/missing-certificate-for-secret/
```
Comment (AndiDog) [inline helm/prometheus-rules/templates/alerting-rules/certificate.management-cluster.rules.yml:29]: This ops-recipe doesn't exist. Is it going to be added @njuettner?

**Pull request merged**